components/Themes/src/MDCFontScheme.h

```diff
@@ -0,0 +1,90 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/**
+ A color scheme comprised of set of primary and secondary colors. Material design guidelines
+ recommend using primary and secondary colors with light and dark color variants.
+ */
+@protocol MDCFontScheme <NSObject>
+
+/** The font-face to be used for Display 1. */
+@property(nonatomic, strong, nullable, readonly) UIFont *display1;
+
+/** The font-face to be used for Display 2. */
+@property(nonatomic, strong, nullable, readonly) UIFont *display2;
```

Comment:
By making them a UIFont we are forcing clients to load them all upfront. If we make them a MDFFontDiskLoader pointer then they can be lazy loaded.

Reply:
Not all of our components will use the larger display sizes.

Reply:
Understood.  We explicitly use UIFont (as opposed to UIFontDescriptor or fontLoader) because we want to know that if UIFont != nil, we will be able to render text with it.
It is totally reasonable that we might have a UseCustomFontForAllValuesScheme object that uses a fontLoader to return a Roboto UIFont for all of the properties.

Reply:
As far as having multiple (possibly unused) fonts such as the larger display sizes.  The pattern is that you would create a fontScheme once, then apply it to multiple / all of you components, similar to how we (currently) do with the ColorThemers.

Reply:
I realized that lazy loading can be don within the object conforming to the protocol.